File: hooks/level.php
#!/usr/bin/php
<?
require_once("boilerplate.php");

if(strlen($args) < 1) {
	$args = $nick;
}

$nick = mysql_escape_string($args);

$a = db("select * from levels where nick like '$nick%' order by length(nick) limit 1");
if(strlen($a[0]['nick']) > 0)
	reply($a[0]['nick'] . ": " . $a[0]['level']);
else
	reply("$nick not found");


?>

File: hooks/level.rb
who = ($args =~ /(\S+)/) ? $1 : $msg.nick
q =
  if (account = Account.name_by_nick(who))
    who = account
    DB[:levels].filter(account: account)
  else
    DB[:levels].filter(nick: who)
  end
if (l = q.first)
  reply "#{who} is #{l[:level]}"
else
  reply "#{who} has no level"
end
